hazelcast/src/main/java/com/hazelcast/map/impl/operation/QueryOperation.java

```diff
@@ -114,6 +111,41 @@ public void run() throws Exception {
         checkPartitionStateChanges(partitionService, partitionStateVersion);
     }
 
+    private void fullTableScan(Collection<Integer> initialPartitions, GroupProperties groupProperties)
+            throws InterruptedException, ExecutionException {
+        if (pagingPredicate != null) {
+            runParallelForPaging(initialPartitions);
+        } else {
+            boolean parallelEvaluation = groupProperties.QUERY_PREDICATE_PARALLEL_EVALUATION.getBoolean();
+            if (parallelEvaluation) {
+                runParallel(initialPartitions);
+            } else {
+                runSingleThreaded(initialPartitions);
+            }
+        }
+    }
+
+    protected void runSingleThreaded(final Collection<Integer> initialPartitions) {
+        RuntimeException storedException = null;
+        MapServiceContext mapServiceContext = mapService.getMapServiceContext();
+        MapContextQuerySupport querySupport = mapServiceContext.getMapContextQuerySupport();
+        for (Integer partitionId : initialPartitions) {
+            try {
+                Collection<QueryableEntry> entries = querySupport.queryOnPartition(name, predicate, partitionId);
+                if (!addQueryEntriesToResult(entries)) {
+                    break;
+                }
+            } catch (RuntimeException e) {
+                if (storedException == null) {
```

Comment:
I'm not sure if you have to safe the exception.

The `getResult()` from the parallel execution has the `FutureUtil.RETHROW_EVERYTHING` exception handling set. If I understood the code correctly an exception is re-thrown immediatly on each `future.get()`. This would be each `querySupport.queryOnPartition()` call in your method. In the parallel execution this would also happen before even a single query entry is added to the result. So it should be safe to do this directly in the for loop.